### PR TITLE
E2E Latency Comparison

### PR DESCRIPTION
**Pull request by manudhundi**
Code to measure E2E latency in Aptos, AVAX-C, Eth based chains, Near, Solana, Sui

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import axios from 'axios';
+
+let VM_URL = "https://gw-c7-2b.cloud.victoriametrics.com";
```
Comment (sitalkedia) [inline common.js]: VM_URL? Let's rename this and also better not to hardcode the default URL.

**Review by sitalkedia — APPROVED** [review]
Minor nit

File: common.js (hunk 1 above)
Comment (manudhundi) [reply common.js]: done

File: README.md [hunk 2]
```diff
@@ -1 +1,30 @@
-# e2e-latency-bench
\ No newline at end of file
+# E2E Latency Comparison Benchmark
+This benchmark measures E2E Latency across several blockchains. The numbers produced by the benchmark represent an end user’s observed latency from submitting a transaction to getting a transaction confirmation back. We compared the latency of coin transfer transactions across multiple blockchain networks in their mainnet environments. Coin transfer is a straightforward and cost-effective transaction type that is widely supported by SDKs.
+
+The benchmark leverages TS SDKs for the respective blockchains. The logic of 'Coin transfer' transaction for a blockchain is present in the corresponding 'index.js' file. Some common helper functions are present in 'common.js'.
+
+## Prerequisites
```
Comment (rustielin) [inline README.md:8]: probably also write that they will need a timeseries database that supports the prometheus remote_write protocol: https://prometheus.io/docs/concepts/remote_write_spec/, as that's what we're using to push metrics from each of the scripts.

File: README.md [hunk 3]
```diff
@@ -1 +1,30 @@
...
+* Aptos: `PING_INTERVAL=900 CHAIN_NAME=mainnet METRICS_URL=<url> METRICS_AUTH_TOKEN=<token> METRICS_TAG=<tag> ACC1_PRIVATE_KEY=<priv_key_sender> ACC2_PRIVATE_KEY=<priv_key_receiver> node index.js`
+* Solana: `PING_INTERVAL=900 CHAIN_NAME=mainnet-beta METRICS_URL=<url> METRICS_AUTH_TOKEN=<token> METRICS_TAG=<tag> ACC1_PRIVATE_KEY=<priv_key_sender> ACC2_PRIVATE_KEY=<priv_key_receiver> COMMITMENT_LEVEL=confirmed node index.js`
+* ETH_BASED_CHAINS (Optimism): `PING_INTERVAL=900 CHAIN_NAME=mainnet METRICS_URL=<url> METRICS_AUTH_TOKEN=<token> METRICS_TAG=<tag> COIN_TRANSFER_LATENCY_METRIC_NAME=e2e_p2p_txn_latency_optimism ACC1_PRIVATE_KEY=<priv_key_sender> ACC1_ADDR=<public_addr_sender> ACC2_ADDR=<public_addr_receiver> URL=https://mainnet.optimism.io TRANSFER_AMT=<amt> node index.js`
+* NEAR: `PING_INTERVAL=900 CHAIN_NAME=mainnet METRICS_URL=<url> METRICS_AUTH_TOKEN=<token> METRICS_TAG=<tag> ACC1_PRIVATE_KEY=<priv_key_sender> ACC1_ID=<public_addr_sender> ACC2_ID=<public_addr_receiver> URL=https://rpc.mainnet.near.org node index.js`
+
+Note: Some env variables are optional, and env variables needed might change based on the script
```
Comment (rustielin) [inline README.md:30]: is it documented somewhere which env vars are optional and which aren't?

File: aptos/p2p_transfer/index.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import { getPrometheusMetricPushPayload, pushPrometheusMetricsToVM, sleepAsync } from './common.js';
+
+const COIN_TRANSFER_LATENCY_METRIC_NAME = "e2e_p2p_txn_latency";
+const COIN_TRANSFER_BUILD_LATENCY_METRIC_NAME = "e2e_p2p_txn_latency_build";
+const COIN_TRANSFER_SUBMIT_LATENCY_METRIC_NAME = "e2e_p2p_txn_latency_sign_and_submit";
+const COIN_TRANSFER_SUCCESS_METRIC_NAME = COIN_TRANSFER_LATENCY_METRIC_NAME + "_success";
+const CHAIN_NAME = process.env.CHAIN_NAME;
+const PING_INTERVAL = process.env.PING_INTERVAL * 1000;
+const CUSTOM_NETWORK = process.env.CUSTOM_NETWORK;
+const INDEXER_URL = process.env.INDEXER_URL;
```
Comment (rustielin) [inline aptos/p2p_transfer/index.js:20]: nit: can we derive the indexer url from the chain name? (for non-custom public networks)

File: common.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            continue;
+        }
+        labelsStrs.push(`${k}="${v}"`);
+    }
+    
+    const allLabelsStr = labelsStrs.join(",");
+    return `${name}{${allLabelsStr}} ${value}`;
+  }
+  
+  async function pushPrometheusMetricsToVM(payloads) {
```
Comment (rustielin) [inline common.js]: nit: rename this to something more generic? like `pushPrometheusMetrics`

File: common.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import axios from 'axios';
+
+const METRICS_URL = process.env.METRICS_URL;
```
Comment (rustielin) [inline common.js:3]: can we pass these in as args rather than env vars for common utilities?

**Review by rustielin — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (manudhundi) [reply README.md:30]: added more examples

File: common.js (hunk 5 above)
Comment (manudhundi) [reply common.js]: done

File: README.md (hunk 2 above)
Comment (manudhundi) [reply README.md:8]: done

**Pull request merged**